app/lib/api/firebase_api.dart

```diff
@@ -0,0 +1,36 @@
+import 'package:firebase_messaging/firebase_messaging.dart';
+
+import '../services/firebase_service.dart';
+import '../services/router_service.dart';
+
+
+class FirebaseApi {
+  final _firebaseMessaging = FirebaseMessaging.instance;
+
+  void handleMessage(RemoteMessage? message) {
+    if (message == null) return;
+
+    router.push('/material');
```

Comment:
why push to `/material`?

have you tried what we can use in `RemoteMessage`, so we can choose how to route it?

Reply:
let's skip `RemoteMessage` since it confuses you
instead, please replace `/material` to `/main` for more general target (more make sense when we have a notification about new quiz)